### PR TITLE
Add MarshalText and UnmarshalText for Duration

### PR DESCRIPTION
**Pull request by metalmatze**
This is mostly needed for https://github.com/alecthomas/kong.
Furthermore it should be useful for others too, as it implements stdlib interfaces.

### CODE REVIEW

**Comment by roidelapluie** [comment]
model.Duration is an internal format, generally kong should use time.Duration. What would be the usecase?

**Comment by metalmatze** [comment]
time.Duration doesn't support anything above hours. Therefore it would need a custom wrapper to support `1w`, `1y` and more. 
We want to add flags that support these time ranges too.
As there's no direct dependency on anything and this change implements two stdlib interfaces I don't see this hurting.

**Comment by roidelapluie** [comment]
I am not sure that I'd like to advise large projects like kong to rely on prometheus common given all the direct and indirect dependencies we have. We have complains like https://github.com/prometheus/common/issues/255 already, and I do not want to amplify these. I agree that the code change is small, but it is also using exported interfaces only, such as this could as well be done downstream in kong with the same amount of code:

`type promDuration model.Duration`

Would that be an option?

**Comment by roidelapluie** [comment]
Okay, can you add a comment about which interface you are implementing?

File: model/time_test.go [hunk 1]
```diff
@@ -157,6 +157,79 @@ func TestParseDuration(t *testing.T) {
...
+			t.Errorf("Unexpected error on input %q", c.in)
+		}
+		if time.Duration(d) != c.out {
+			t.Errorf("Expected %v but got %v", c.out, d)
+		}
+		expectedString := c.expectedString
+		if c.expectedString == "" {
+			expectedString = c.in
+		}
+		text, _ := d.MarshalText() // MarshalText returns hardcoded nil
```
Comment (roidelapluie) [inline model/time_test.go:226]: For the future it would have been great to check for error, even if we only return nil.

**Comment by roidelapluie** [comment]
Thanks!!

**Pull request merged**